### ISSUE TITLE
Room list filtering is painfully slow

### ISSUE THREAD

**Issue by turt2live** [vector-im/element-web#14750]

**Comment by ordo-ordinvm** [comment]
I had this same problem using Ubuntu 16.04 LTS 4.4.0-108-generic. Room list is blank for several minutes while filtering takes place. For some reason if you select one of the rooms in the community from the history bar at the top during filtration, it filters instantly.

**Comment by turt2live** [comment]
@ordo-ordinvm I think that'll be fixed in the next release, as it sounds unrelated to this issue.

**Comment by gsouquet** [comment]
I got a hold of Matthew's performance profile after the first round of optimisation I did.

That led to matrix-org/matrix-react-sdk#5908, but outside of that there not a whole lot of simple optimisation that one can achieve here.

The next logical step would be to look into virtualising the list but that does involve changing the structure of the list and has quite a larger scope compared to those low hanging fruit I've been tackling.

We've agreed with @jryans to leave this one aside and get back to it at a later stage
